### PR TITLE
Align documentation with recommendation of Linux Documentation Project

### PR DESCRIPTION
**Pull request by gustafn**
CLA: trivial

This change applies the recommendation of the Linux Documentation Project
to the documentation files of OpenSSL. Additionally, util/find-doc-nits
was updated accordingly.

The change follows a suggestion of  mspncp  on https://github.com/openssl/openssl/pull/12370

##### Checklist

- [x] documentation is added or updated

### CODE REVIEW

**Comment by t-j-h** [comment]
This is not a trivial change IMHO. It requires a CLA. I would otherwise approve it.

**Comment by mspncp** [comment]
Hmm... I thought it was consensus to consider spelling corrections as not copyrightable?

**Comment by t-j-h** [comment]
These aren't spelling corrections ... and it does come to the scope of a change moving from trivial to non-trivial.

File: util/find-doc-nits [hunk 1]
```diff
@@ -551,43 +551,54 @@ sub functionname_check {
...
 my %preferred_words = (
+    '16bit'         => '16-bit',
     'a.k.a.'        => 'aka',
     'bitmask'       => 'bit mask',
     'builtin'       => 'built-in',
    #'epoch'         => 'Epoch', # handled specially, below
     'file name'     => 'filename',
     'file system'   => 'filesystem',
     'host name'     => 'hostname',
+    'host name'     => 'hostname',
```
Comment (slontis) [inline util/find-doc-nits:562]: Look like a duplicate?

File: util/find-doc-nits [hunk 2]
```diff
@@ -551,43 +551,54 @@ sub functionname_check {
...
     'i-node'        => 'inode',
     'lower case'    => 'lowercase',
     'lower-case'    => 'lowercase',
+    'manpage'       => 'man page',
+    'non-blocking'  => 'nonblocking',
+    'non-default'   => 'nondefault',
+    'non-empty'     => 'nonempty',
+    'non-negative'  => 'non-negative',
```
Comment (slontis) [inline util/find-doc-nits:570]: ??

File: util/find-doc-nits [hunk 3]
```diff
@@ -551,43 +551,54 @@ sub functionname_check {
...
     'set-GID'       => 'set-group-ID',
-    'setgid'        => 'set-group-ID',
     'set-UID'       => 'set-user-ID',
+    'setgid'        => 'set-group-ID',
     'setuid'        => 'set-user-ID',
-    'super user'    => 'superuser',
-    'super-user'    => 'superuser',
+    'sub-system'    => 'subsystem',
     'super block'   => 'superblock',
+    'super user'    => 'superuser',
```
Comment (slontis) [inline util/find-doc-nits:590]: move this down one.

File: util/find-doc-nits [hunk 4]
```diff
@@ -551,43 +551,54 @@ sub functionname_check {
...
+    'super-user'    => 'superuser',
+    'system port'   => 'privileged port',
     'time stamp'    => 'timestamp',
     'time zone'     => 'timezone',
     'upper case'    => 'uppercase',
     'upper-case'    => 'uppercase',
     'useable'       => 'usable',
-    'userspace'     => 'user space',
     'user name'     => 'username',
+    'user name'     => 'username',
```
Comment (slontis) [inline util/find-doc-nits:600]: duplicate again?

**Comment by gustafn** [comment]
Although i share the opinion of mspncp (how can the application of publicly accepted spelling rules create a copyright infringement claim?) i know as main contributor of a larger long-term projects with many contributors that e.g. license updates can be a pain in such project. So i have filed an ICLA via email to the requested address.

It is my understanding that
- sooner or later you will have my email-address on file such that
- a "CLA: trivial" or some other explicit reference to a "CLA" will be needed in the future.

All requested changes are addressed in the updated branch

**Comment by t-j-h** [comment]
Remove the CLA: trivial header and once your CLA is registered (I saw it come in) then you just close and open the case to trigger a recheck and it will be good to proceed. Thanks for taking that step quickly - the change and your CLA are very much appreciated.

**Comment by gustafn** [comment]
> Remove the CLA: trivial header 

In order change the previous commit message, i did a "git reset" in the feature branch to the change before my first commit, did a new commit without the "CLA: trivial" and a "git push -f". I hope, this is what was intended.

> you just close and open the case to trigger a recheck .
Ok, i will close and reopen this pull request after some time (e.g. this evening).

**Comment by slontis** [comment]
> In order change the previous commit message, i did a "git reset
FYI: 
If you only need to fix the last commit like this you can just do
```
git commit --amend
```
It still requires the force push..

For multiple commits you can use
```
git rebase -i master
```
And then edit the commit line you need to change to use 'r' (reword)

**Comment by mspncp** [comment]
@gustafn it seems like your email address hasn’t been added to the database yet. Whoever does it will reopen you ticket to kick the CLA bot, so you need not worry about it.

**Comment by mattcaswell** [comment]
Close/reopen to kick CLA bot.

**Comment by mspncp** [comment]
Merged to master in 490c87110cc1fd673604fa1b94d5538f9fd852bb. Thank you @gustafn!

**Pull request closed without merging**